clang/test/CIR/CodeGen/tbaa.cpp

```diff
@@ -0,0 +1,262 @@
+// modified from clang/test/CodeGen/tbaa.cpp
+// with clangir enabled and some debugging info
+// if disable debuggging info is supported by clangir
+// the debugging info can be removed
+// and keep the test case the same with clang/test/CodeGen/tbaa.cpp
+
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -O1 -no-struct-path-tbaa -disable-llvm-passes %s -emit-llvm -o - | FileCheck %s
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -O1 -disable-llvm-passes %s -emit-llvm -o - | FileCheck %s -check-prefixes=PATH,OLD-PATH
+// Test TBAA metadata generated by front-end.
+//
+// NO-TBAA-NOT: !tbaa
+
+typedef unsigned char uint8_t;
+typedef unsigned short uint16_t;
+typedef unsigned int uint32_t;
+typedef unsigned long long uint64_t;
+typedef struct
+{
+   uint16_t f16;
+   uint32_t f32;
+   uint16_t f16_2;
+   uint32_t f32_2;
+} StructA;
+typedef struct
+{
+   uint16_t f16;
+   StructA a;
+   uint32_t f32;
+} StructB;
+typedef struct
+{
+   uint16_t f16;
+   StructB b;
+   uint32_t f32;
+} StructC;
+typedef struct
+{
+   uint16_t f16;
+   StructB b;
+   uint32_t f32;
+   uint8_t f8;
+} StructD;
+
+typedef struct
+{
+   uint16_t f16;
+   uint32_t f32;
+} StructS;
+typedef struct
+{
+   uint16_t f16;
+   uint32_t f32;
+} StructS2;
+
+uint32_t g(uint32_t *s, StructA *A, uint64_t count) {
+// CHECK-LABEL: define{{.*}} i32 @_Z1g
+// CHECK: store i32 1, ptr %{{.*}}, align 4, !dbg !{{.*}}, !tbaa [[TAG_i32:!.*]]
```

Comment:
> it would be nice if that can also be fixed to prevent this annoying extra outputs!

I would like to take on the issue https://github.com/llvm/clangir/issues/793 

In my original plan, I intended to add debug info after completing support for TBAA.

Reply:
Awesome, for some reason your username is not showing up for me to assign to you, but I just gave you more repo permission to now be able to assign it yourself, let me know if doesn't work